### ISSUE TITLE
Some Questions about the paper

### ISSUE THREAD

**Issue by JIANG-CX** [autonomousvision/factor-fields#2]
Nice work!
I have questions about two details of the paper:
1. I am confused about why Instant-NGP is considered to employ a "basis representation" and why it is described as a "global field".
2. What do "axis-aligned bias" and "sensitivity to the orientation of the coordinate system" mean in the context of "orthogonal coordinate transformation"?
Thanks.

**Comment by apchenstu** [comment]
Hi, good questions.
1.  the instant-NGP shares the feature embeddings globally at the fine level due to the hash function, that's one feature embedding representing multiple spatial points of the whole input domain, if unflatten the 1D embeddings to the input signal domain, you can observe some basis patterns.
2. you can consider the target signal as a high-dimensional matrix, the different orientations can result in significant matrix rank, where a high-rank matrix requires more decomposition components, while a low-rank matrix doesn't. And the component number are generally pre-defined and may be insufficient when the matrix is high rank, resulting in worse performance. For example, the lego scene in the TensoRF repo may have worse results if you apply a random rotation on the scene coordinate.

**Comment by JIANG-CX** [comment]
> Hi, good questions.
> 
> 1. the instant-NGP shares the feature embeddings globally at the fine level due to the hash function, that's one feature embedding representing multiple spatial points of the whole input domain, if unflatten the 1D embeddings to the input signal domain, you can observe some basis patterns.
> 2. you can consider the target signal as a high-dimensional matrix, the different orientations can result in significant matrix rank, where a high-rank matrix requires more decomposition components, while a low-rank matrix doesn't. And the component number are generally pre-defined and may be insufficient when the matrix is high rank, resulting in worse performance. For example, the lego scene in the TensoRF repo may have worse results if you apply a random rotation on the scene coordinate.

Thanks for your reply.
I am curious if the term "basis" in the paper refers to patterns that will be spatially repeated.
Thanks.

**Comment by JIANG-CX** [comment]
> Hi, good questions.
> 
> 1. the instant-NGP shares the feature embeddings globally at the fine level due to the hash function, that's one feature embedding representing multiple spatial points of the whole input domain, if unflatten the 1D embeddings to the input signal domain, you can observe some basis patterns.
> 2. you can consider the target signal as a high-dimensional matrix, the different orientations can result in significant matrix rank, where a high-rank matrix requires more decomposition components, while a low-rank matrix doesn't. And the component number are generally pre-defined and may be insufficient when the matrix is high rank, resulting in worse performance. For example, the lego scene in the TensoRF repo may have worse results if you apply a random rotation on the scene coordinate.

Thank you very much for your response. I apologize for still being confused about these two questions, as I understand it might take up your valuable time. I appreciate your patience and understanding.
1. I am curious if the term "basis" in the paper refers to patterns that will be spatially repeated.
2. Does this mean that if a random rotation is applied to the scene coordinates before training, the results will be worse? If this is indeed the case, how can we determine the number of components in advance?
3. Based on your second question, can I infer that EG3D employs a tri-plane approach for matrix decomposition, which may be inadequate when dealing with matrices of high rank?
Thanks.

**Comment by apchenstu** [comment]
> 1. I am curious if the term "basis" in the paper refers to patterns that will be spatially repeated.

would be better to refer me to the paper section so that I can locate the related context

> 2. applied to the scene coordinates before training, the results will be worse? If this is indeed the case, how can we determine the number of components in advance?

yes, it's not possible before reconstructing the signal, the component number is a hyperparameter

> 3. Based on your second question, can I infer that EG3D employs a tri-plane approach for matrix decomposition, which may be inadequate when dealing with matrices of high rank?

yeah, TensoRF also shares similar limitation, btw, the number of factors and the number of components is kind of different.

**Comment by JIANG-CX** [comment]
> > 1. I am curious if the term "basis" in the paper refers to patterns that will be spatially repeated.
> 
> would be better to refer me to the paper section so that I can locate the related context
> 
> > 2. applied to the scene coordinates before training, the results will be worse? If this is indeed the case, how can we determine the number of components in advance?
> 
> yes, it's not possible before reconstructing the signal, the component number is a hyperparameter
> 
> > 3. Based on your second question, can I infer that EG3D employs a tri-plane approach for matrix decomposition, which may be inadequate when dealing with matrices of high rank?
> 
> yeah, TensoRF also shares similar limitation, btw, the number of factors and the number of components is kind of different.

I have figured out these questions. Thank you for your answers. This is great work!

**Comment by FantasticOven2** [comment]
Hi,
regarding to the question  

> _What do "axis-aligned bias" and "sensitivity to the orientation of the coordinate system" mean in the context of "orthogonal coordinate transformation"?_

We have a paper study the axis-aligned bias and here's the link for your reference:
[https://brentyi.github.io/tilted/](url)

**Comment by JIANG-CX** [comment]
> Hi, regarding to the question
> 
> > _What do "axis-aligned bias" and "sensitivity to the orientation of the coordinate system" mean in the context of "orthogonal coordinate transformation"?_
> 
> We have a paper study the axis-aligned bias and here's the link for your reference: [https://brentyi.github.io/tilted/](url)

Thanks, I will read it.
